web-app/js/datasetExplorer/datasetExplorer.js

```diff
@@ -780,32 +779,91 @@ Ext.onReady(function () {
+    function loadPlugin(pluginName, scriptsUrl, bootstrap, legacy) {
+        var def = jQuery.Deferred();
+        function loadResources() {
+            loadResourcesByUrl(
+                pageInfo.basePath + scriptsUrl,
+                function() {
+                    def.resolve.apply(def, arguments);
+                })
+                .fail(function() { def.reject.apply(def, arguments); });
+        }
+
+        // if it's a legacy call (hardcoded in datasetExplorer.js),
+        // we need to check whether the plugin is present
+        if (legacy) {
+            jQuery.post(pageInfo.basePath + '/pluginDetector/checkPlugin',
+                {pluginName: pluginName })
+                .done(function(data) {
+                if (data === 'true') {
+                        loadResources()
+                } else {
+                        console.log("Plugin " + pluginName + " not active");
+                        def.reject('not active');
+                }
+                })
+                .fail(function() {
+                    console.log('Could not determine whether plugin ' + pluginName +
+                            ' is active');
+                    def.reject.apply(def, arguments);
+                });
+        } else {
+            loadResources();
+        }
 
-        loadPlugin('smartR', "/SmartR/loadScripts", function () {
-            resultsTabPanel.add(smartRPanel);
+        pluginPromises.push({
+            promise: def.promise(),
+            bootstrap: bootstrap,
         });
+    }
 
-        // ----
+    /* load the leegacy hardcoded tabs */
+    loadPlugin('dalliance-plugin', '/Dalliance/loadScripts', function () {
+        loadDalliance(resultsTabPanel);
+    }, true);
+    loadPlugin('transmart-metacore-plugin', '/MetacoreEnrichment/loadScripts', function () {
+        loadMetaCoreEnrichment(resultsTabPanel);
+    }, true);
+
+    /* load the tabs registered with the extension mechanism */
+    (function loadAnalysisTabExtensions() {
+        GLOBAL.analysisTabExtensions.forEach(function(tabExtension) {
+        loadPlugin(null, tabExtension.resourcesUrl, function () {
+            (window[tabExtension.bootstrapFunction])(resultsTabPanel, tabExtension.config);
+            });
+        });
+    })();
+
+    // wait for all the tab scripts to be loaded, and only then run their bootstrap code
+    // this ensures that they are added in a predictable order
+    (function runTabBootstrapCode() {
+        var def = jQuery.Deferred();
+        var n = pluginPromises.length;
+        // cannot use .when() because we don't want to fail when
+        // one of the plugins fails to load
+        pluginPromises.forEach(function(promiseAndBootstrap) {
+            promiseAndBootstrap.promise.always(function() {
+                if (--n == 0) {
+                    def.resolve();
+    }
+            });
+        });
 
-        if(!Math.log10){ Math.log10 = function (x) { return Math.log(x) / Math.log(10); }}
```

Comment:
Is this intentionally removed? I think it was required to support some browser version.